packages/jest-worker/src/workers/NodeThreadsWorker.ts

```diff
@@ -137,6 +150,55 @@ export default class ExperimentalWorker implements WorkerInterface {
     this._resolveExitPromise();
   }
 
+  private async monitorHeartbeatError() {
+    let error = new Error(
+      `Test worker was unresponsive for 10 seconds. There was no inspector connected so we were unable to capture stack frames before it was terminated.`,
+    );
+    if (this._options.inspector) {
+      error = new Error(
+        `Test worker was unresponsive for 10 seconds. There was an inspector connected so we were able to capture stack frames before it was terminated.`,
+      );
+      this._options.inspector.on('Debugger.paused', (message: any) => {
+        const callFrames = message.params.callFrames.slice(0, 20);
+        for (const callFrame of callFrames) {
+          const loc = callFrame.location;
+
+          const columnNumber = loc.columnNumber;
+          const lineNumber = loc.lineNumber;
+          const url = callFrame.url;
+
+          const name = callFrame.scopeChain[0].name;
+
+          // TODO: process error with stack trace
+          console.log({
+            columnNumber,
+            lineNumber,
+            name,
+            url,
+          });
+        }
+      });
+
+      await new Promise((resolve, reject) => {
+        this._options.inspector?.post('Debugger.pause', (err: Error) => {
+          if (err === null) {
+            resolve();
+          } else {
+            reject(err);
+          }
+        });
+      });
+    }
+    // @ts-expect-error: adding custom properties to errors.
+    error.type = 1;
```

Comment:
`1`? should we add a `name`?

Reply:
Should we add another `PARENT_MESSAGE_ERROR` ?

```ts
// types.ts
export const PARENT_MESSAGE_HEARTBEAT_ERROR: 5 = 5;

export type PARENT_MESSAGE_ERROR =
  ...
  | typeof PARENT_MESSAGE_HEARTBEAT_ERROR;
```

Reply:
That makes sense to me

Reply:
Is this actually worth it though? It's not a message that is actually ever sent is it? It's just to  set a property on an error that can further identify the error but that I don't think we really need?
I think declaring messages that are not actually messages ever being passed between parent and child is confusing